client/apps/edit/components/admin/components/autocomplete.coffee

```diff
@@ -0,0 +1,45 @@
+React = require 'react'
+ReactDOM = require 'react-dom'
+async = require 'async'
+request = require 'superagent'
+sd = require('sharify').data
+_ = require 'underscore'
+{ div, label, input, form, span } = React.DOM
+AutocompleteList = require '../../../../../components/autocomplete_list/index.coffee'
+
+module.exports = React.createClass
+  displayName: 'AdminAutocompleteList'
+
+  componentDidMount: ->
+    @setupAutocomplete(@props.field)
+
+  componentWillUnmount: ->
+    $(@refs['autocomplete']).each (i, ref) ->
+      ReactDOM.unmountComponentAtNode(ref)
+
+  setupAutocomplete: (field) ->
+    fieldSingle = field
+      .replace('_ids', '')
+      .replace('_artsy', '')
+      .replace('_about', '')
+      .replace('_programming', '')
+    placeholder = "Search #{fieldSingle} by name..."
+    fieldSingle = fieldSingle.replace('auction', 'sale')
+    fieldPlural = fieldSingle + 's'
+    url = "#{sd.ARTSY_URL}/api/v1/match/#{fieldPlural}?term=%QUERY"
+    new AutocompleteList $(@refs['autocomplete'])[0],
+      url: url
+      placeholder: placeholder
+      filter: (res) -> for r in res
+        { id: r._id, value: r.name }
+      selected: (e, item, items) =>
+        @props.onChange field, _.pluck items, 'id'
+      removed: (e, item, items) =>
+        @props.onChange field, _.without(_.pluck(items, 'id'),item.id)
+      idsToFetch: @props.article.get field
+      fetchUrl: (id) -> "#{sd.ARTSY_URL}/api/v1/#{fieldSingle}/#{id}"
+      resObject: (res) ->
+        id: res.body._id, value: res.body.name
```

Comment:
Curious about these too, is it possible to get them in the initial render function so it's clear that we're including a react component?